src/explore-education-statistics-admin/src/pages/release/pre-release/components/__tests__/PreReleaseUserAccessForm.test.tsx

```diff
@@ -36,11 +36,11 @@ describe('PreReleaseUserAccessForm', () => {
 
       const row1Cells = within(rows[1]).getAllByRole('cell');
 
-      expect(row1Cells[0]).toHaveTextContent('test1@education.gov.uk');
+      expect(row1Cells[0]).toHaveTextContent('test1@test.com');
```

Comment:
This was already here, but it'd be good to improve the performance of these tests by moving `getAllByRole` and `getAllByRole` to after awaits. In this case you could check for the text that appears above the table in the await then the rest after it, eg:
```
await waitFor(() => {
expect(screen.getByText('These people will have access to a preview of the release 24 hours before the scheduled publish date.').toBeInTheDocument();
})
 const rows = screen.getAllByRole('row');
 expect(rows).toHaveLength(3);
  etc...
```

There are a few other tests in this file that could have a similar improvement.

Reply:
Thanks, I've made some improvements in https://github.com/dfe-analytical-services/explore-education-statistics/pull/3429/commits/1c17ad19878460fdf4f69d1cb0cc29fa6669d3ba. Let me know if I've missed anything 🙂 .